### DISCUSSION TITLE
Request for Dynamic named index Support in Data Collection

### DISCUSSION THREAD

**Discussion by SaeedNikmehr** [spatie/laravel-data#339] [Ideas]
I was hoping to utilize the current package in a manner where I can dynamically change the currency codes, such as USD or AUD, in the data collection for my request object. The format I am aiming for is as follows:
```
"pricing": {
                    "USD": {
                        "prefix": "$",
                        "suffix": " USD",
                    },
                    "AUD":{
                        "prefix": "$",
                        "suffix": " AUD",
                   }

```
Unfortunately, I have not been able to find any solutions within the current package to accommodate this requirement. I would appreciate it if you could guide me in the right direction or suggest a solution to meet my needs. Thank you for your time and efforts.

**Comment by rubenvanassche** [comment]
How would this look in code?
